tests/mvrf/test_mgmtvrf.py

```diff
@@ -14,60 +15,81 @@
 logger = logging.getLogger(__name__)
 
 
+def restore_config_db(duthost):
+    # Restore the original config_db to override the config_db with mgmt vrf config
+    duthost.shell("mv /etc/sonic/config_db.json.bak /etc/sonic/config_db.json")
+
+    # Reload to restore configuration
+    config_reload(duthost)
+
+
 @pytest.fixture(scope="module", autouse=True)
 def setup_mvrf(duthost, testbed, localhost):
     """
     Setup Management vrf configs before the start of testsuite
     """
-    logging.info("Configure mgmt vrf")
-    global var
-    global mvrf
-    mvrf = True
-    var = {}
-    var["dut_ip"] = duthost.setup()["ansible_facts"]["ansible_eth0"]["ipv4"]["address"]
-    var["ptf_ip"] = testbed["ptf_ip"]
-    var["filename"] = "README.md"
-    duthost.command("sudo config vrf add mgmt")
-    SONIC_SSH_REGEX = "OpenSSH_[\\w\\.]+ Debian"
-
-    verify_show_command(duthost)
+    # Backup the original config_db without mgmt vrf config
+    duthost.shell("cp /etc/sonic/config_db.json /etc/sonic/config_db.json.bak")
+
+    try:
+        logger.info("Configure mgmt vrf")
+        global var
+        global mvrf
+        mvrf = True
+        var = {}
+        var["dut_ip"] = duthost.setup()["ansible_facts"]["ansible_eth0"]["ipv4"]["address"]
+        var["ptf_ip"] = testbed["ptf_ip"]
+        var["filename"] = "README.md"
+        duthost.command("sudo config vrf add mgmt")
+        SONIC_SSH_REGEX = "OpenSSH_[\\w\\.]+ Debian"
+        verify_show_command(duthost)
+    except Exception as e:
+        logger.error("Exception raised in setup, exception: {}".format(repr(e)))
+        restore_config_db(duthost)
+        pytest.fail("Configure mgmt vrf failed, no test case will be executed. Teardown will not be executed either.")
```

Comment:
I don't know if this is accurate. We restored config DB, which is sort of teardown, right? :P

Reply:
Right. Updated to "Code after 'yield' will not be executed either".